### ISSUE TITLE
Please update, hand over, or close Quickdocs

### ISSUE THREAD

**Issue by phoe** [quickdocs/quickdocs#46]
As #45 mentions, Quickdocs has been stale and was not updated for almost a whole year now. This causes increasing confusion and issues and I worry that it negatively affects the Common Lisp world as a whole, as small as it is.

For instance, today I fixed a bug in someone else's environment who downloaded a project from quickdocs instead of git. This way, they downloaded a 2019 version, and then wondered why it did not work on their modern Linux/Lisp system. #45 links another instance of related issues

I propose the following ways forwards:
* update Quickdocs with the recent Quicklisp dists and keep it updated in the future,
* allow other people to maintain Quickdocs and keep it up to date,
* close the project and make it unavailable from the Internet.

Any of the three options is IMO better than letting Quickdocs stay stale, as it is right now.

**Comment by Hexstream** [comment]
I fully support this motion. @fukamachi has been extensively, breathtakingly, inexplicably irresponsive (and perhaps irresponsible) and this must come to a stop. It would also be great if he could provide at least an outline of an explanation for why this sorry state has been perpetuating.

**Comment by mdbergmann** [comment]
Also agree. Should be taken down.
What is the big benefit over quickref?

**Comment by Hexstream** [comment]
> Also agree. Should be taken down.

This seems like the most drastic and destabilizing option, but we definitely seem to have a consensus that one of the 3 options must be implemented.

> What is the big benefit over quickref?

I think it's a bit of an anachronistic question, as Quickdocs came first. I think there is room for a plurality of options. The big problem here is the extended lack of updates resulting in decrepitude.

**Comment by mdbergmann** [comment]
> This seems like the most drastic and destabilizing option

I'm not sure. I think it confused more than it is useful. But I might be mistaken.

> I think there is room for a plurality of options.

Sure. But I don't understand the purpose of it. It's called "Quickdocs" but it provides more than just the docs, right?

**Comment by Hexstream** [comment]
> I'm not sure. I think it confused more than it is useful. But I might be mistaken.

Well, I must say I was relieved [when the Common Lisp Directory finally died](https://twitter.com/HexstreamSoft/status/1259518989576876032) after years of abject decrepitude. The Quickdocs situation is not *quite* as bad yet, but it is much more popular, so there *is* significant harm in the current situation.

Also, [I'm personally not really a big fan of automated documentation extraction services in general](https://twitter.com/HexstreamSoft/status/1311028894804439041).

**Comment by mdbergmann** [comment]
> Also, I'm personally not really a big fan of automated documentation extraction services in general.

That's perfectly fine for me. In Java it's a big relief to create API documentation this way. The IDE can pick this up automatically, etc. It's standard. Same for .NET, Erlang/Elixir, OCaml.
The problem in Common Lisp is that there is no standard (or community standard) for this. There is no way to add markup, or define what should be extracted and what not.

**Comment by Hexstream** [comment]
I've found that documentation written by hand absolutely can reach a high level of quality, while autogenerated documentation absolutely cannot.

**Comment by mdbergmann** [comment]
Probably it depends on what and who the documentation is for.

**Comment by Hexstream** [comment]
I prefer documentation written for humans, which autogenerated documentation seemingly never is.

**Comment by mdbergmann** [comment]
Why not. Of course it is.
But inline documentation is not meant to transport the full picture, or concepts.

**Comment by Hexstream** [comment]
[I tweeted a polite appeal to Fukamachi to come resolve this issue](https://twitter.com/HexstreamSoft/status/1311336174364037126), since merely opening issues on GitHub has often been quite ineffective in getting his attention, historically.

**Comment by digikar99** [comment]
I think both together are better than either - though probably this discussion belongs elsewhere - human-written documentation can convey the bigger picture; but sometimes, one just wants to know what a particular construct is/does. And perhaps, with a bit of standardization, both of these can be combined-automated as well: extract the human-made big-picture.md from the repository and append it with the docstrings. But, yeah, even docstrings need a bit of standardization if hyperlinks to other mentioned symbols are to be automatically generated.

I didn't get the quickdocs view of preserving the code structure rather than arranging symbols in alphabetical order. Is it like, when an author writes a library, the order in which the constructs are written conveys a better meaning than simple alphabetical order? If so, I guess I seem to prefer an explicit big-picture.md.

I found quickdocs at least invaluable in one way, to find out how oft-used a library is; but more generally, for quicklisp metadata; is there some way to obtain quicklisp metadata?

**Comment by Hexstream** [comment]
> I think both together are better than either [...]

I've found [this mental model](https://twitter.com/HexstreamSoft/status/1296100536790515712) useful.

> I found quickdocs at least invaluable in one way, to find out how oft-used a library is;

Absolutely! This is my favorite Quickdocs feature. [Here is a nice example, see "Required by" at the bottom.](http://quickdocs.org/closer-mop/)

> but more generally, for quicklisp metadata; is there some way to obtain quicklisp metadata?

Unfortunately, Quicklisp is [notoriously undocumented](https://twitter.com/HexstreamSoft/status/1085446845000765440) (but otherwise [appreciated](https://twitter.com/HexstreamSoft/status/1309635519362543617)).

**Comment by Hexstream** [comment]
I recognize that Fukamachi may be extremely busy for legitimate personal reasons.

As such, personally, I would be satisfied if Fukamachi publicly committed to implementing one of the 3 proposed options by the end of the year.

**Comment by linnilbobo** [comment]
I guess that he caused an error during a certain maintenance process, which caused the website to fail to display the updated content. If it needs to be repaired, it may have to close the entire website for several hours. In order to maintain stability, he chose not to make changes. All the information is frozen on that day. Maybe he needs help more instead of accusing him of being lazy to update

**Comment by mdbergmann** [comment]
The download numbers, what do they represent. Only what has been download from Quickdocs?

The 'requires/depends on' thingy is kind of nice.
I've already disqualified myself, so I can say it: Quickdocs is ugly as shit.
The front page categorization is completely incomplete and I don't see the concept or the bigger picture in all of this.
However, this doesn't mean Quickdocs can't be improved.
And yet, there is no bigger picture.

**Comment by Hexstream** [comment]
@linnilbobo Is this pure speculation? But personally, at this point I wouldn't mind a lot if the site was completely unavailable for several hours or even a few days if this is what it took to resolve the issue.

Our crisis of confidence is particularly compounded by Fukamachi's extended silence on this issue.

**Comment by linnilbobo** [comment]
Just my guess

**Comment by Hexstream** [comment]
@mdbergmann

> The download numbers, what do they represent. Only what has been downloaded from Quickdocs?

I'm not sure, I'm curious about this too.

> I've already disqualified myself, so I can say it: Quickdocs is ugly as shit.

Actually, I think Quickdocs is probably one of the most visually appealing Common Lisp sites, which is one reason why I am not that eager to see it die (unless it turns out to be the best option).

**Comment by phoe** [comment]
@linnilbobo I can't say anything about laziness since I don't know the reasons for why the project has been left unmaintained for a long time, especially since there's thousands of good reasons for doing so. Still, the current situation of the project is that Quickdocs shows up very high in SEO results, e.g. for [`cl-lzma`](https://www.google.com/search?q=cl-lzma) it's on top, above the original github repository for the project. The same happens for multiple other Common Lisp projects and it only gets worse over time.

Whatever is the reason for this lack of updates, the final results of that is problems, increasing confusion, and proliferation of old versions of autogenerated documentation. This is why I consider it of high importance to do *something* about this state of matters, regardless of what exactly it is, and I need to say that leaving a highly-utilized and important project like Quickdocs silently go stale for full nine months is nothing that I could call responsible behavior.

**Comment by mdbergmann** [comment]
> I think Quickdocs is probably one of the most visually appealing Common Lisp sites

OK. That tells tales. I hear me telling to my boss, that function is more important. But visuals are important, too, and a certain amount of aesthetics massively lowers barriers.
I'm not a CSS expert, otherwise I could help. But the selection of fonts, sizes, general structure can massively be improved in Quicklisp.
Take a look at this: https://hex.pm and choose the top download: jason package.

**Comment by linnilbobo** [comment]
I used the wrong words, please forgive me. 
I also agree to ask him to come out and give an explanation

**Comment by Hexstream** [comment]
@mdbergmann I just think that it's fairly contextually irrelevant to criticize one of the most visually appealing Common Lisp sites for allegedly not being visually appealing enough when the much more pressing problem *is in its function*.

Also note that `hex.pm` is not a Common Lisp site, and is thus quite unrelated to my quote.

**Comment by mdbergmann** [comment]
@Hexstream 
Well, I couldn't disagree more. The Quickdocs page is as visually unappealing as it could be.
The reason I linked hex.pm is to look at the visuals.
Visuals are visuals and have nothing to do whatsoever with this or that programming language.
But visuals are what people see first when they look at Common Lisp sites.
And btw. this page: https://common-lisp.net is visually appealing. Why not put all this under one umbrella?

**Comment by phoe** [comment]
Also, regarding the original post: I apologize. My tone was unnecessarily harsh, and I could have worded the same content more skillfully and with less anger.

**Comment by linnilbobo** [comment]
https://common-lisp.net/phub

**Comment by Hexstream** [comment]
@phoe Strongly disagree. Your original post was pitch-perfect, as far as I can tell.
We need more frank communication in this community, and you are contributing to it.

**Comment by mdbergmann** [comment]
> https://common-lisp.net/phub

@linnilbobo 
For some reason I can't find mine. Is this linked to Quickdocs?
Where is the documentation.
I'm sorry. But man, what a mess.

**Comment by phoe** [comment]
@mdbergmann Likely this only lists projects hosted on the common-lisp.net infrastructure. Are your projects hosted there?

**Comment by linnilbobo** [comment]
> Is this linked to Quickdocs?
no, they are different

**Comment by mdbergmann** [comment]
@phoe 
No, it's probably not hosted there.

But you see, that's part of the problem I guess. There is too much stuff that just creates confusion.
And I apologise. I'm getting a bit aroused here.
You know, I really like Common Lisp.

**Comment by stylewarning** [comment]
Eitarow—maintainer of Quickdocs and author of a zillion libraries—recently had a child and was just discharged from a long stay at the hospital. Give him a break.

Some of you in this thread are being completely socially irresponsible. Be polite. Maybe the maintainers of your favorite projects don’t have the same cozy disposition as you in life.

Best of all, offer help. If the help isn’t received, go do something else.

**Comment by phoe** [comment]
@Hexstream No. My original post was unnecessarily aggressive, entitled, and lacked respect towards Fukamachi and Quickdocs altogether. It was pointed out to me, and it is what I apologize for.

I can understand that you generally enjoy such style of communicating with people and that you call it "frankness", but the contents of my original post are not up to my personal standards of online discussion.

**Comment by mdbergmann** [comment]
> have the same cozy disposition as you in life

Hope all is well and healthy. Having kids is really the best thing in life.

**Comment by Hexstream** [comment]
@phoe Somewhat agree, but I do think that an artificial lack of "negativity", mostly enforced through various malicious means, is one of our biggest problems in the Common Lisp community.

**Comment by phoe** [comment]
@stylewarning Thanks for the reality check and for the information.

I guess that these are the questions that I should have asked from the start: 

* What is required to get Quickdocs updated once, as a hotfix for their current situation?
* What is required to keep Quickdocs automatically updated in the future, e.g. via some CI mechanism?
* How can someone from outside (e.g. me) help with this process or share the burden with Fukamachi?

**Comment by Hexstream** [comment]
@stylewarning Thank you very much for this crucial information, but 9 months of increasingly problematic stagnation without any communication still seems a bit excessive in spite of these considerations (just a quick note would have done the trick), and I did pre-emptively propose that [we should give him some flexibility to account for possible personal circumstances](https://github.com/quickdocs/quickdocs/issues/46#issuecomment-701512613), but I am sure we were expected to guess what happened out of thin air, as always...

**Comment by mdbergmann** [comment]
My additional question would be (not fully related, but I ask it anyway because there is currently the context for it):

- can the style of common-lisp.net be also used in Quickref and Quckdocs? The styles are completely different.
- can Quickdocs be hosted under common-lisp.net as a subdomain similarly as Quickref?
- and finally, when Quickdocs is again up-to-date and stabilized, can Quickref be removed? What is the purpose of having both?

**Comment by Hexstream** [comment]
@mdbergmann [Here are some of my personal insights about common-lisp.net.](https://twitter.com/HexstreamSoft/status/1250142510267224065)

My conclusion is that everyone should consolidate on GitHub, as far as possible.

**Comment by stylewarning** [comment]
This entire thread is noise. It should be closed, started afresh with substantive requests and feedback, with all the new facts considered. Commentary from the sidelines should be avoided, especially if they pertain to “The Community” or “Common Lisp At Large”. A bug report or feature request doesn’t need to devolve into social philosophy of open source.

**Comment by phoe** [comment]
Done.

**Comment by Hexstream** [comment]
@stylewarning Profoundly disagree, obviously, but soft censorship is indeed one of our most harmful yet popular memes.

**Comment by stylewarning** [comment]
@Hexstream Not every forum was designed as a channel for your personal broadcasts. You have plenty of outlets to share your opinions. Being “on topic” isn’t “censorship”.

**Comment by phoe** [comment]
> @stylewarning Thank you very much for this crucial information, but 9 months of increasingly problematic stagnation without any communication still seems a bit excessive in spite of these considerations (just a quick note would have done the trick), and I did pre-emptively propose that we should give him some flexibility to account for possible personal circumstances, but I am sure we were expected to guess what happened out of thin air, as always...

@Hexstream, I hope that you are aware that your exquisitely elongated expedient of extremely elevated, elaborate and exasperated expression comes off as high amounts of irony and/or sarcasm, which can (and will) be read as passive aggression. This directly contributes to how your words are received, and then directly contributes to people's reaction to them.

Like, look, I can compress this whole above post by 50% *and* remove all the seemingly-aggressive stuff at the same time.

> @stylewarning Thanks, but 9 months of total silence still seems like a lot in this case (just a quick note would have done the trick). I suggested that Fukamachi might have personal reasons for this, but we had no information whatsoever.

Same stuff with this "soft censorship" that you mentioned. What's being censored here? Who? The first post of #47 lists, IMO, the most important questions of this whole thread so far. Anything and everything else, really, is not strictly on-topic, and I'd prefer to focus on the topic that is Quickdocs being out of date. (This includes the Quickdocs CSS discussion that is IMO worth its own separate ticket... once #47 gets resolved.)

**Comment by qu1j0t3** [comment]
> I fully support this motion. @fukamachi has been extensively, breathtakingly, inexplicably irresponsive (and perhaps irresponsible) and this must come to a stop. It would also be great if he could provide at least an outline of an explanation for why this sorry state has been perpetuating.

Big words 

![Screen Shot 2020-09-30 at 3 29 45 PM](https://user-images.githubusercontent.com/424616/94731367-a5cf3380-0332-11eb-94f4-8228ff7ae6c4.png)

**Comment by Hexstream** [comment]
@stylewarning

> This entire thread is noise.

This is literally the most harmful thing that has been said in this entire thread, and frankly it's pretentious and disrespectful to suggest it. We obviously did not reach peak on-topicness, but many substantive and at least arguably related points have been made. A little noise does not completely invalidate all other value.

> It should be closed, started afresh with substantive requests and feedback, with all the new facts considered.

This just needlessly splits the discussion, thereby *increasing* confusion.

> Commentary from the sidelines should be avoided, especially if they pertain to “The Community” or “Common Lisp At Large”.

You are continuing a long tradition of violently ignoring Common Lisp community concerns, which almost invariably means that you will see these discussions where you don't want to see them. My preferred way to prevent this from happening, as always, would be to simply resolve these problems, which would take comparatively little effort.

> A bug report or feature request doesn’t need to devolve into social philosophy of open source.

And it usually doesn't, and if it does once in a while, I think that's fine.

> Not every forum was designed as a channel for your personal broadcasts.

Quite correct, but given the fairly relentless (mostly "soft") censorship I have been a target of for close to a decade, I see it as a near inevitability that a certain form of compensation is useful, since my writings are largely prevented from being seen where they belong.

> You have plenty of outlets to share your opinions.

And I am grateful for it, but they are all afflicted by the soft censorship issue.

> Being “on topic” isn’t “censorship”.

Actually [I strongly value on-topicness](https://chat.hexstreamsoft.com/rules/), but sometimes there are overriding concerns.

**Comment by Hexstream** [comment]
@phoe You are under the delusion that there is some way I could use to express myself that would be significantly better received, which is unfortunately not the case, if we are talking about in any way preserving the substance of what I say.

**Comment by linnilbobo** [comment]
In any case, I think we’re on the right track. We all want the Lisp community to flourish, so we can stop arguing about this problem. My suggestion is to add a bulletin board in the README section of the project showing the update plan, and the reason for the pause update.

In addition, we can look out the window of the Moon, today is the mid-autumn Festival.
The roundest days of this year are a symbol of family reunion.
Fate brings us together here, let us celebrate together.

**Comment by Hexstream** [comment]
@qu1j0t3 Actually, this is trivially explained. My usual modus operandi is to work on a project privately before an initial release, at which point I squash all my private commits into one "Initial commit" for public release. I tend to do this even for very ambitious projects. So I may well work on a big project for more than a month, which then results in a single commit.

I could explain and demonstrate this much more extensively, but to cut things short, you may like to [review my work](https://www.hexstreamsoft.com/) for a much more enlightened impression of my productivity.

But thank you for your [concern](https://pics.me.me/concern-30338992.png).

**Comment by phoe** [comment]
@Hexstream To add, [the comment that you have just made](https://github.com/quickdocs/quickdocs/issues/47#issuecomment-701591193) in #47 contributes **nothing** to the real issue of keeping Quickdocs up to date. You could have not posted there, or even post in here, instead of proudly stating your opinion about some fact unrelated to the original topic and disregarding the fact that this opinion contributes nothing the original issue of getting Quickdocs up to date and instead forces people who want to read about the original issue of Quickdocs being out of date to spend their energy on reading your opinion instead, which is boldly stated, but unrelated to Quickdocs being out of date.

IMO, this is a good illustration of what @stylewarning said about not every forum being suitable for your personal broadcasts.

Since you like pointing out genuine problems, this behavior above is a textbook example of [attention seeking](https://en.wikipedia.org/wiki/Attention_seeking), a well-described behavior commonly [associated](https://en.wikipedia.org/wiki/Histrionic_personality_disorder) with [several](https://en.wikipedia.org/wiki/Borderline_personality_disorder) prominent [personality](https://en.wikipedia.org/wiki/Narcissistic_personality_disorder) disorders, as well as what is commonly called "childishness". And the fact that you often exhibit such behavior is a genuine problem.

For instance, this kind of behavior makes the places where you make such posts a worse place, because people are spending their energy on reading unrelated stuff, instead of doing meaningful work that they have wanted to read about. This wastes people's time and energy. This wastes your time and energy, like, imagine all the stuff you could have done and could be doing right now if only you didn't feel the compulsive need to post unrelated stuff and then defend your compulsive reasons and needs for posting that.

This is a fiftieth or so time I've saw you do this on the Internet. This is a genuine problem. As a member of the Common Lisp community, honestly, I'm fed up with it.

Your posts across the Internet are literally dripping with entitlement (e.g. where you are *entitled* to have your work linked to other parts of the Lisp world, such , and that you are *entitled* to call some people the [Common Lisp Mafia](https://github.com/exercism/common-lisp/issues/206#issuecomment-421618037) and then [post](https://twitter.com/HexstreamSoft/status/639937501822615552) about its omnipotence; or talk about a grand conspiracy about you) and pretentiousness (e.g. where you tweet that ["one day, Hexstream versus Zach Beane will be studied in universities"](https://twitter.com/HexstreamSoft/status/1308954558853513218), as if any university in the world cared about either you or Zach Beane enough to devote course material to either or both of you; or, again, that multiple people care enough to form an actual [conspiracy](https://twitter.com/HexstreamSoft/status/1041671580957466624) to destroy the Common Lisp community by "censoring" you out of it), and then straight out martyrology (where you claim you are [the only one](https://twitter.com/HexstreamSoft/status/1041678388656857088) seeing a massive gaping issue with the Common Lisp community that almost everyone else is simply blind to for whatever reason; or where you need to justify the fact that [your "exposing" of Xach does not have the effect you imagine](https://twitter.com/HexstreamSoft/status/1041920287875493888), and so you throw crap over the whole Common Lisp community by calling it, as a whole, "mentally ill").

It's gone to the point where you have, yourself, [publicly defamed mfiano](https://twitter.com/HexstreamSoft/status/1267093005271433218), [publicly defined dkochmanski](https://twitter.com/HexstreamSoft/status/1171975874276679680), [publicly accused stassats of "having his hands tied"](https://twitter.com/HexstreamSoft/status/1043671920649142272) by, I infer, the Common Lisp mafia, [written countless Twitter rants about Xach](https://twitter.com/HexstreamSoft/status/1054828493467541510), [created a PSA](https://www.hexstreamsoft.com/#psa-about-zach-beane-the-sociopath) on your website listing those rants, and registered the domain https://xach.exposed in order to link to them. All of this, while staying completely oblivious to the fact that you're sending truckloads of passive and active aggression all around you in all directions, and *this* is the real thing that drives people away from you.

And, lo and behold, this thread, as many others, has evolved into you expressing your opinions of being permanently censored and attacked by various hostile means and actors from within the Common Lisp community, and then you use those opinions as rationale for putting up your personal and greatly elaborate self-defense, where you link to previous examples of this behaviour that were, simularly, stimulated by your provocation, entitlement, and pretentiousness.

Seriously. This was supposed to be a Quickdocs thread. And then you arrived, and started to derail it from [this](https://github.com/quickdocs/quickdocs/issues/46#issuecomment-701557192) point onwards by means of showing off your opinion that was off-topic, and then starting to defend your reasons for posting that opinion, and then accusing other posters of being against you, and then, help, help, you're being oppressed.

I seriously don't know what to say about all of this. Uh, I hope that you really start taking proper care of yourself, for your own personal inner peace that you obviously don't have with the amount of aggression gushing out of you, and for the peace of the Common Lisp community that could use fewer aggression pouring out of you?

**Comment by phoe** [comment]
@Hexstream Regarding my "delusion": yes, there is such a way. I just posted it above. Look:

> @stylewarning Thanks, but 9 months of total silence still seems like a lot in this case (just a quick note would have done the trick). I suggested that Fukamachi might have personal reasons for this, but we had no information whatsoever.

It preserves the whole essence of the original post. It preserves nothing, or little, of the original's aggression. From my point of view, this last post of yours means that you either refuse to see what I posted above or you admit that the removed aggression *is* the substance of what you say. I don't know which variant is more trobulesome.

**Comment by Hexstream** [comment]
@phoe Obviously I disagree with most of your points, but I applaud you for producing such a post exhibiting the kind of frankness that will obviously be required to [get ourselves out of our pit of despair](https://trends.google.com/trends/explore?date=all&q=Common%20Lisp).

A few points demand swift an unequivocal refutal, however. (Fortunately, you have conveniently grouped them together.)

> It's gone to the point where you have, yourself, publicly defamed mfiano, publicly defined dkochmanski, publicly accused stassats of "having his hands tied"

What I said [about mfiano](https://twitter.com/HexstreamSoft/status/1267093005271433218) and [about jackdaniel](https://blog.hexstream.xyz/Daniel-Kochma%C5%84ski-jackdaniel-rushed-to-send-a-MALICIOUS-LIBELOUS-AND-FRAUDULENT-complaint-to-Cloudflare/) is 100% true, and none of them have even attempted to refute my allegations (I don't count jackdaniel's few weak, slander-ridden attempts), even though they absolutely both did see them, because they know they are true. jackdaniel can go ahead and post a serious refutal of [my article about him](https://blog.hexstream.xyz/Daniel-Kochma%C5%84ski-jackdaniel-rushed-to-send-a-MALICIOUS-LIBELOUS-AND-FRAUDULENT-complaint-to-Cloudflare/). As for mfiano, pretty much the first thing he did was get me banned from Lisp discord for no reason because he couldn't handle me, so I don't expect a response. (And there is much more to say about it, but this is already pretty long.)

As for stassats, I did not *accuse* him of [having his hands tied](https://twitter.com/HexstreamSoft/status/1043671920649142272), I ***excused*** him for it, which is a pretty big semantic difference.

> Regarding my "delusion": yes, there is such a way. I just posted it above. Look:

This is just a micro-optimization, it would not really have changed anything in the grand scheme of things, and if it had meant that you hadn't posted your great post above, then it would have been greatly harmful.

> or you admit that the removed aggression *is* the substance of what you say

Aggression is never the substance of what I say, but that doesn't necessarily mean that removing it would improve anything, especially if it was just plain old repression.

**Comment by phoe** [comment]
> Obviously I disagree with most of your points,

That's your right and your problem at this point, I've made the points I've wanted to make.

> I applaud you for producing such a post 

No problem. As much as I'm fed up with your delusion, I respect your work and you as a person, and I hope that you'll eventually find your peace and stop needing to thrash around as wildly as you are right now. (Tip: that depends mostly on you and the state of your own inner world, not on things and people that happen outside.)

> exhibiting the kind of frankness

Frankly, a big part of your problem with the outside world is inside you, and therefore the outside world is unable to fix it. The only thing it can do is trying to point you in the direction of the actual problem, but it can't change stuff for you.

> This is just a micro-optimization, it would not really have changed anything in the grand scheme of things, and if it had meant that you hadn't posted your great post above, then it would have been greatly harmful.

No, it's not a microoptimization, it is a pretty big *transformation*: it removes aggression from a post and therefore makes it much more readable and accessible to other people. Make that times ten, a whole thread of yours may become easy to read for other people. Make that times hundred, it's possible that people will stop finding your overall daily behavior troublesome. Make that times thousand, it's possible that you'll solve the issue of yourself getting "censored" or "cancelled" out of multiple Internet forums that are as fed up of you as I am at the moment, e.g. the Lisp Discord that you mention.

This is because, contrary to what you seem to think, you are *not* entitled to being on the Lisp Discord and having everyone listen to whatever you say, just like I am not entitled to being there and having everyone listen to whatever I say. Being a part of some community is a privilege, and so far you have been using that privilege to annoy lots of people and, in result, get yourself banned from some places, leave a bad taste in multiple mouths, and finally have some people want to cut all sorts of connections with you because of your behavior - e.g. via refusing to publish your work, deleting existing links to your work, refusing to associate themselves with work that you do as a person, and so on, and so on.

There, look, it's caused your behavior and not some grand conspiracy. Wherever you go with such kind of attitude, you're going to start the same flamewars, and the Common Lisp Mafia™, just like you successfully observe, will follow you wherever you are - simply because you shall end up creating it yourself by antagonizing people against you with your childishness.

That's actually an optimistic point of view, you know? Stop the cause, and the effect will eventually cease, too. If the issue is somewhere in your behavior, then you're in full control and capability of changing it for having an easier life and a better Lisp community, starting with your. If the issue is some mafia acting outside, not only you have no control over it and you're doomed to befall whatever plots it plans for you, but you also spend lots of worthwhile time and energy thrashing against whatever you perceive to be of its making.

> because he couldn't handle me

Why would he be required to handle you, or why would *anyone* be required to handle you? This is a literal definition of entitlement. Other people owe you nothing beside basic human respect, which I'm exhibiting right now by trying to tell you what I see without any beating around the bush. This respect doesn't include unconditionally letting you into any sort of group, community, or chat.

> Truth is an absolute defense against the charge of slander.

The truth is that your behavior contains lots of pretentious attention-seeking nuisance conveyed via passive-aggressive style of writing. Because of that, don't be surprised that people, including people that you spend tons of time writing about on Twitter, simply show you the door and/or move you through it by force. And they have done both.

> Aggression is never the substance of what I say

Even if it isn't the substance, it is the medium, which causes the same effect. Aggression is aggression, and you are not entitled to any kind of audience where you can proudly flaunt it around for attention. Even more so, you are not entitled to having the Common Lisp community turn into such kind of audience for you, despite how much you seem to attempt to do so on Twitter.

**Comment by Hexstream** [comment]
@phoe Thank you for so nakedly sharing your perspective, which I found slightly simplistic, but still very interesting.
You are a great asset to the Common Lisp community, and [I really wish I could (safely) fund you](https://twitter.com/HexstreamSoft/status/1267523948390031365).

> Other people owe you nothing beside basic human respect

It is all I really ask, but fairly seldom what I receive, I believe.
I am sure some would call this "entitlement", though.

**Comment by Hexstream** [comment]
@phoe Since you [mentioned](https://github.com/quickdocs/quickdocs/issues/46#issuecomment-701620431) [histrionic personality disorder](https://en.wikipedia.org/wiki/Histrionic_personality_disorder), **have you considered that *you may have it*?
Because *this sounds exactly like you:***

* **Constant seeking of reassurance or approval**
* **Excessive sensitivity to criticism or disapproval**
* A need to be the center of attention
* Low tolerance for frustration or delayed gratification
* **Rapidly shifting emotional states that may appear superficial or exaggerated to others**
* Tendency to believe that relationships are more intimate than they actually are
* Making rash decisions
* Blaming personal failures or disappointments on others
* **Being easily influenced by others, especially those who treat them approvingly**
* **Being overly dramatic and emotional**
* **Influenced by the suggestions of others**

**Comment by thephoeron** [comment]
Yes. Histrionic is definitely the word I would use to describe @phoe.

In any case, I'm also available to assist with the maintenance and hosting of Quickdocs.

**Comment by phoe** [comment]
@thephoeron @Hexstream Thanks for the insight, I'll consider it.

**Comment by thephoeron** [comment]
You're welcome, @phoe. And remember, no judgement here. I've got PTSD, and can get pretty temperamental myself, the difference being that I need to be triggered.

There's a lot of really crappy people in the Start-Up world, that think they have a right to use my PTSD against me, triggering me on purpose for the sake of abuse or manipulation or their own control issues.

We're not like them. We're Lisp Hackers, and we've got each other's backs.
